spk/rutorrent/src/wizard_templates/base_install_uifile.json

```diff
@@ -0,0 +1,100 @@
+        "items": [
+            {
+                "type": "combobox",
+                "desc": "Please select a volume to use for the download folder",
+                "subitems": [
+                    {
+                        "key": "wizard_download_volume",
+                        "desc": "volume name",
+                        "displayField": "display_name",
+                        "valueField": "volume_path",
+                        "editable": false,
+                        "mode": "remote",
+                        "api_store": {
+                            "api": "SYNO.Core.Storage.Volume",
+                            "method": "list",
+                            "version": 1,
+                            "baseParams": {
+                                "limit": -1,
+                                "offset": 0,
+                                "location": "internal"
+                            },
+                            "root": "volumes",
+                            "idProperty": "volume_path",
+                            "fields": [
+                                "display_name",
+                                "volume_path"
+                            ]
+                        }
+                    }
+                ]
+            },
+            {
+                "type": "textfield",
+                "desc": "Download shared folder (using the volume chosen above)",
+                "subitems": [
+                    {
+                        "key": "wizard_download_dir",
+                        "desc": "Download location",
+                        "defaultValue": "downloads",
+                        "validator": {
+                            "allowBlank": false,
+                            "regex": {
+                                "expr": "/^[\\w _-]+$/",
+                                "errorText": "Subdirectories are not supported."
+                            }
+                        }
+                    }
+                ]
+            },
+            {
+                "desc": "If the specified share does not exist, it will be created. You can use an existing share by specifying the name of the folder."
+            },
+            {
+                "type": "textfield",
+                "desc": "Watch a directory for torrent files and add them to ruTorrent. Leave empty to disable",
+                "subitems": [
+                    {
+                        "key": "wizard_watch_dir",
+                        "desc": "Watch directory",
+                        "validator": {
+                            "allowBlank": true,
+                            "regex": {
+                                "expr": "/^\\\/volume[0-9]+\\\//",
+                                "errorText": "Path should begin with /volume?/ with ? the number of the volume"
+                            }
+                        }
+                    }
+                ]
+            },
```

Comment:
IMHO this share must be separated to volume and share name as above for the download location
OR you use the same volume as above (we did not test whether the creation of multiple shared folders on different volumes for the same package is supported by DSM).
OR you create the watch folder as a sub directory of the download shared folder, then you have to adjust at least the validation regex.

Reply:
I would add, why not simply follow the exact same directory tree as deluge and enforce that by default such as:
```
downloads/
├── complete
├── incomplete
└── watch
```
There's a dsm6-7 "compatible" method to create the folders from deluge `service-setup.sh` script:
```
# Create download directories
install -m 0775 -o ${EFF_USER} -g ${GROUP} -d "${incomplete_folder}"
install -m 0775 -o ${EFF_USER} -g ${GROUP} -d "${complete_folder}"
install -m 0775 -o ${EFF_USER} -g ${GROUP} -d "${watch_folder}"
```

Reply:
We should use a different watch folder for each package, as we do not want to download a watched file by all watching applications (and it will depend on whether an application removes the files in the watch folder when download starts vs when the download is completed).

Reply:
Good point, maybe worth defaulting it to `watch-<appname>` then?  I could make the changes to deluge while i have an opened pr.

Reply:
Actually I would propose to have a combination of

1. A checkbox allowing for usage of a watch directory (disabled by default)
2. An input component defaulted to the "watch" subdirectory

WDYT ?

I am actually not sure that rutorrent supports the complete / incomplete logics.

Reply:
Sounds good to me.

As for the complete vs incomplete directories, if I recall you have to set the default directory to incomplete and use the move upon completion feature so it, well, move the resulting torrent to the complete folder once over.

One thing i noticed is the incredible performance difference between deluge vs rutorrent.  Deluge is numerous time more performant ... There may be config options to fine tune rutorrent...

Traveling this week so my availability may vary.

Reply:
Sorry for the delays here.

First things first: my idea of having a checkbox is actually not feasible it seems with regards to the current Javascript framework which is used to handle the installer logics (short version: can't listen for a checkbox change to actually change visibility of a component, I can only do that on page changes)

I'm not (at all) an expert in rtorrent configuration, I may be having a look at this but it may take some time.

And for the performance part I must confess I don't use rutorrent (apart for testing, I'm more of a transmission user) and only had a crack at it to unblock some python logics so I can't really argue with you here :)

Will try having a look today.

Reply:
@smaarn, is this concern regarding the UI wizard logic now resolved?

Reply:
@mreid-tt no it's not. The wizard option is actually a dead-end and the actual rutorrent options need being tested.

I must confess I didn't consider investing more time in "normalizing" those options. TBH as a Transmission user I found it puzzling on my last upgrade to see the paths change so not sure it's something we want to force, is it ?

I mean, for an initial installation, it's arguable, but I'd say that for an existing installation it can be seen as a little bit "invasive" for a policy, is it not ?

Reply:
> First things first: my idea of having a checkbox is actually not feasible it seems with regards to the current Javascript framework which is used to handle the installer logics (short version: can't listen for a checkbox change to actually change visibility of a component, I can only do that on page changes)

@smaarn, as a side note, I was working on a more robust uninstall UI wizard for ownCloud ([uninstall_uifile.sh](https://github.com/SynoCommunity/spksrc/blob/85c9a6c8609332caaea9295f26b1582ada6c739b/spk/owncloud/src/wizard/uninstall_uifile.sh)) and after going down a trial-and-error rabbit-hole, I can confirm the issue with checkboxes and component visibility. I did get it to partially listen for a checkbox to make another item visible using a validation function but there were two major issues:

1. The actual checkbox itself when clicked didn't do anything
2. The checkbox label did trigger the change but only after clicking elsewhere (remove focus)

Either way, this was not an ideal situation. I did however find a use-case to apply a validation function to a text field and based on the input, enable checkboxes (or other fields) as required. From a UX perspective it works because the user would want to click one of the empty checkboxes after entering a path and this action would make the text field loose focus – immediately enabling the checkboxes before the click is registered. I remembered this when I wrapped up and thought I'd share my findings.